### ISSUE TITLE
[Web][Safari]: Unable to enter text into Textfield on latest master and beta.

### ISSUE THREAD

**Issue by drexel-ue** [flutter/flutter#96411]
## Steps to Reproduce

1. Run web application on. mac device using `flutter run -d web-server --web-hostname localhost --web-port 5000 --release`
  - Can also be deployed via firebase hosting
2. Open web application using Safari
3. Attempt to type in any textfield

**Expected results:** 
Textfield should gain focus and accept text upon keyboard button presses

**Actual results:**
Textfield gains focus and does not accept text upon keyboard button presses

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}


class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Demo(),
    );
  }
}

class Demo extends StatefulWidget {
  const Demo({Key? key}) : super(key:key);
  
  @override
  _DemoState createState() => _DemoState();
}

class _DemoState extends State<Demo> {
  @override build(BuildContext context) {
    return const Scaffold(
      body: Material(
        child: Center(
          child: TextField(),
        ),
      ),
    );
  }
}
```

</details>

<details>
  <summary>Logs</summary>

```
flutter analyze                                                                       ─╯
Analyzing url_landing_site...                                           

   info • The value of the field '_lastUser' isn't used • lib/app/site.dart:34:9 •
          unused_field
   info • 'textSelectionHandleColor' is deprecated and shouldn't be used. Use
          TextSelectionThemeData.selectionHandleColor instead. This feature was deprecated
          after v1.26.0-18.0.pre. • lib/app/theme.dart:28:5 • deprecated_member_use
   info • Avoid `print` calls in production code •
          lib/backend/pay/pay_method_button.dart:182:13 • avoid_print
   info • Avoid `print` calls in production code •
          lib/backend/pay/pay_method_button.dart:194:13 • avoid_print
   info • Avoid `print` calls in production code •
          lib/backend/pay/pay_method_button.dart:197:15 • avoid_print
   info • Avoid `print` calls in production code •
          lib/backend/pay/pay_method_button.dart:205:15 • avoid_print
   info • Don't access members with `this` unless avoiding shadowing •
          lib/models/api.dart:29:46 • unnecessary_this
   info • The import of 'package:flutter/gestures.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/screens/_common/footer.dart:1:8 • unnecessary_import
   info • The import of 'package:flutter/rendering.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/screens/_common/footer.dart:3:8 • unnecessary_import
   info • The import of 'package:flutter/rendering.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/screens/home/home.dart:2:8 • unnecessary_import
   info • The import of 'package:url_landing_site/models/models.dart' is unnecessary because
          all of the used elements are also provided by the import of
          'package:url_landing_site/backend/backend.dart' •
          lib/screens/home/page_home_logged_in.dart:6:8 • unnecessary_import
   info • The import of 'package:flutter/rendering.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/screens/other/faq.dart:2:8 • unnecessary_import
   info • The value of the field '_mode' isn't used •
          lib/screens/subscribe/payment_type_selection_page.dart:22:46 • unused_field
   info • The import of 'package:url_landing_site/models/models.dart' is unnecessary because
          all of the used elements are also provided by the import of
          'package:url_landing_site/backend/backend.dart' •
          lib/screens/subscribe/subscription_dialog.dart:12:8 • unnecessary_import
   info • The import of 'package:url_landing_site/models/models.dart' is unnecessary because
          all of the used elements are also provided by the import of
          'package:url_landing_site/backend/backend.dart' • lib/screens/video/video.dart:3:8
          • unnecessary_import
   info • The import of 'package:flutter/widgets.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/utils/formatting.dart:2:8 • unnecessary_import
   info • The import of 'package:flutter/widgets.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/utils/media_query_extension.dart:24:8 • unnecessary_import
   info • The import of 'package:flutter/foundation.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/widgets/anchored_tooltip.dart:1:8 • unnecessary_import
   info • The import of 'package:flutter/foundation.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/widgets/cursor.dart:1:8 • unnecessary_import
   info • The import of 'package:flutter/rendering.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/widgets/cursor.dart:4:8 • unnecessary_import
   info • The import of 'package:flutter/foundation.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/widgets/iframe.dart:1:8 • unnecessary_import
   info • The operand can't be null, so the condition is always true •
          lib/widgets/site_dialog.dart:414:41 • unnecessary_null_comparison
   info • The import of 'package:flutter/foundation.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/widgets/svg.dart:3:8 • unnecessary_import
   info • The import of 'package:flutter/rendering.dart' is unnecessary because all of the
          used elements are also provided by the import of 'package:flutter/material.dart' •
          lib/widgets/text_link.dart:2:8 • unnecessary_import

```

```
╰─ flutter doctor -v                                                                     ─╯
[✓] Flutter (Channel beta, 2.9.0-0.1.pre, on macOS 12.1 21C52 darwin-x64, locale en-US)
    • Flutter version 2.9.0-0.1.pre at /Users/ikkeshpack/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8f1f9c10f0 (4 weeks ago), 2021-12-14 13:41:48 -0800
    • Engine revision 234aca678a
    • Dart version 2.16.0 (build 2.16.0-80.1.beta)
    • DevTools version 2.9.1

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/ikkeshpack/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • ANDROID_HOME = /Users/ikkeshpack/Library/Android/sdk
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] IntelliJ IDEA Community Edition (version 2021.2.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 60.1.4
    • Dart plugin version 212.5080.8

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 97.0.4692.71

• No issues found!


```

</details>

**Comment by darshankawar** [comment]
Thanks for the report. It seems the current behavior is occuring on latest beta and master, while it works properly on latest stable (2.8.1):

1. On beta and master: Running `flutter run -d chrome` and then opening the link on Safari (I am on Catalina OS), the textfield gets the focus but doesn't show anything being typed.

2. On latest stable: Running the `flutter run -d chrome`  or `flutter run -d web-server --web-hostname localhost --web-port 5000 --release` and opening the localhost link on Safari brower gets the focus and shows the input being typed.

<img width="869" alt="Screenshot 2022-01-11 at 2 06 22 PM" src="https://user-images.githubusercontent.com/67046386/148909520-18d2ec96-df8b-485f-9d35-7bb958cd8c0f.png">


<details>
<summary> beta, master flutter doctor -v </summary>

```

[✓] Flutter (Channel master, 2.9.0-1.0.pre.316, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.9.0-1.0.pre.295 at
      /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 79025d4567 (9 minutes ago), 2022-01-10 23:40:18 -0500
    • Engine revision b5a9cfd537
    • Dart version 2.16.0 (build 2.16.0-144.0.dev)
    • DevTools version 2.9.2
    • PATH =
      /usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/Library/Apple/usr/binpplicat
      ions/Software_Setup/flutter/bin:/Users/dhs/triagemagic/bin/:/Users/dhs/doc
      uments/fluttersdk/flutter/bin:/emulator:/tools:/tools/bin:/platform-tools:
      /Users/dhs/.pub-cache/bin
    
[✓] Android toolchain - develop for Android devices (Android SDK version 30)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.3
    • ANDROID_HOME = /Users/dhs/Library/Android/sdk
    • Java binary at: /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.    

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Users/dhs/Library/Application Support/JetBrains/Toolbox/apps/AndroidStudio/ch-0/202.7486908/Android
      Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)        

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[[✓] Connected device (4 available)
    • Darshan's iphone (mobile)  • 21150b119064aecc249dfcfe05e259197461ce23 •
      ios            • iOS 14.4.1 18D61
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729     •
      ios            • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                    •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                                   •
      web-javascript • Google Chrome 93.0.4577.82


• No issues found!

[✓] Flutter (Channel beta, 2.9.0-0.1.pre, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-GB)
    • Flutter version 2.9.0-0.1.pre at /Users/dhs/documents/fluttersdk/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 8f1f9c10f0 (4 weeks ago), 2021-12-14 13:41:48 -0800
    • Engine revision 234aca678a
    • Dart version 2.16.0 (build 2.16.0-80.1.beta)
    • DevTools version 2.9.1

[!] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    ! Flutter recommends a minimum Xcode version of 13.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] VS Code (version 1.62.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.21.0

[✓] Connected device (3 available)
    • iPhone 12 Pro Max (mobile) • A5473606-0213-4FD8-BA16-553433949729 • ios
      • com.apple.CoreSimulator.SimRuntime.iOS-14-3 (simulator)
    • macOS (desktop)            • macos                                •
      darwin-x64     • Mac OS X 10.15.4 19E2269 darwin-x64
    • Chrome (web)               • chrome                               •
      web-javascript • Google Chrome 97.0.4692.71

! Doctor found issues in 1 category.




```

</details>

**Comment by TahaTesser** [comment]
This regression is introduced in https://github.com/flutter/flutter/commit/23c20d702badf669ac63e1777314da45419f46db

**Comment by mdebbar** [comment]
cc @LongCatIsLooong

**Comment by LongCatIsLooong** [comment]
It looks like on desktop safari the input element was never put into the DOM. Working on a pull request to fix that.

**Comment by TahaTesser** [comment]
Closed revert PR based on the above comment, thanks.
I wonder if we're1 testing text input in desktop Safari so this kinda regression doesn't resurface in the future

**Comment by tomgilder** [comment]
Is this going to end up on stable? Feels like a hot fix candidate, as it'll totally break editing text on a major browser?

**Comment by elliotrtd** [comment]
This is marked as a severe regression and was identified 28 days ago yet still has not been put in to stable. 

Issues like this are honestly what make Flutter frustrating and whenever I am asked about it by other developers I have to give plenty of warning. Web has been supposedly stable for a while now, yet I'm going to have to rebuild my web app with the master branch risking whatever other bugs might exist there just to be able to enter text - something I would say is a pretty basic requirement.

**Comment by elliotrtd** [comment]
Oh hallelujah! It looks like this has finally been fixed with Stable version 2.10.1
